apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1207,7 +1216,44 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
     :ok = Queries.add_old_interfaces(db_client, new_state.device_id, old_introspection)
     :ok = Queries.remove_old_interfaces(db_client, new_state.device_id, readded_introspection)
 
-    # TODO: handle triggers for interface minor updates
+    interface_minor_updated_map =
+      Enum.map(old_minors, fn {iface, old_minor} ->
+        old_major = Map.fetch!(state.introspection, iface)
+        {iface, old_major, old_minor}
+      end)
+      |> Enum.filter(fn {iface, old_major, _old_minor} ->
+        Enum.member?(db_introspection_map, {iface, old_major})
```

Comment:
SInce it's a map shouldn't this be `Map.has_key?`?

Reply:
`db_introspection_map` is a name-major map, e.g. `%{"com.example.TestObject" => 1}`, the key would be just interface name which is too little for the minor updated trigger

Reply:
```
        Map.get(db_introspection_map, iface) == old_major
```